connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteThreadRunner.scala

```diff
@@ -162,6 +162,18 @@ private[connect] class ExecuteThreadRunner(executeHolder: ExecuteHolder) extends
             s"${executeHolder.request.getPlan.getOpTypeCase} not supported.")
       }
 
+      if (executeHolder.observations.nonEmpty) {
+        val observedMetrics = executeHolder.observations.map { case (name, observation) =>
```

Comment:
I'm confused here.
It seems we already create and send observed metrics response at `SparkConnectPlanExecution.handlePlan`.

Reply:
The observed metrics sent in `SparkConnectPlanExecution.handlePlan` is now for the case of `input.isStreaming || executeHolderOpt.isEmpty`; otherwise, they will be sent here.

Reply:
Got it.